File: python/ABSTRACT_FACTORY_DECORATOR/ABSTRACT_FACTORY_DECORATOR/reservacion/ReservacionSencilla.py
from ABSTRACT_FACTORY_DECORATOR.reservacion.DatosFuncion import DatosFuncion
from ABSTRACT_FACTORY_DECORATOR.reservacion.DatosCompra import DatosCompra
from ABSTRACT_FACTORY_DECORATOR.reservacion.DatosCliente import DatosCliente
from ABSTRACT_FACTORY_DECORATOR.reservacion.Reservacion import Reservacion


class ReservacionSencilla(Reservacion):

    def __init__(self, datos) -> None:
        datosCliente  = DatosCliente()
        datosCliente.setDatosCliente(datos[0],datos[1],datos[2],datos[3],datos[4])
        datosCompra   = DatosCompra()
        datosCompra.setDatosCompra(datos[5],datos[6])
        datosFuncion = DatosFuncion()
        datosFuncion.setDatosFuncion(datos[7],datos[8],datos[9])
        super().__init__([datosCliente,datosCompra,datosFuncion])
    
    def mostrarReservacion(self):
        super().mostrarReservacion() 

File: php/ABSTRACT_FACTORY_DECORATOR/reservacion/ReservacionSencilla.php
<?php

require_once 'Reservacion.php';
require_once 'DatosCliente.php';
require_once 'DatosCompra.php';
require_once 'DatosFuncion.php';

class ReservacionSencilla extends Reservacion {

    public $datosCliente;
    public $datosCompra;
    public $datosFuncion;

    public function __construct($datos) {
        $this->datosCliente = new DatosCliente();
        $this->datosCompra = new DatosCompra();
        $this->datosFuncion = new DatosFuncion();

        $this->datosCliente->setDatosCliente($datos["nombre"], $datos["apellidoP"], $datos["apellidoM"], $datos["edad"],
                $datos["tipoCliente"]);
        $this->datosCompra->setDatosCompra($datos["numTarjeta"], $datos["fecha"]);
        $this->datosFuncion->setDatosFuncion($datos["asiento"], $datos["hora"], $datos["dia"], $datos["mes"],
                $datos["anio"], $datos["pelicula"]);
        parent::__construct($this->datosCliente, $this->datosCompra, $this->datosFuncion);
    }

    public function mostrarReservacion() {
        return parent::mostrarReservacion();
    }

}
